.github/workflows/quality-assurance.yml

```diff
@@ -0,0 +1,86 @@
+name: quality-assurance
+
+on:
+  pull_request:
+
+env:
+  python-version: "3.9"
+  rust-version: "1.61.0"
+  node-version: "16"
+
+jobs:
+  check-quality-assurance:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
We should pin github action with commit hash instead of tag.

Otherwise an attacker getting access to the action's repo can modify the tag and execute arbitrary code which is able to exfiltrate the secrets we use in our CI (in our case typically api token to upload to snap and pypi).

However it's your lucky day given we already have all the code to implement this ready: 😄 
- https://github.com/touilleMan/godot-python/blob/master/.github/workflows/build.yml
- https://github.com/touilleMan/godot-python/blob/6adbe46550c90c0f676b8835e854eddd1b654a97/.pre-commit-config.yaml#L19-L26
- https://github.com/touilleMan/godot-python/blob/master/misc/pin_github_actions.py

Reply:
done for pinned actions version, for the pre-commit hook, I'll do this in another PR